File: feature/services/src/main/java/com/kotlin/learn/feature/services/sample/MyForegroundService.java
package com.kotlin.learn.feature.services.sample;

import android.app.Notification;
import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.app.Service;
import android.content.Intent;
import android.os.IBinder;
import android.util.Log;

import androidx.annotation.Nullable;

import com.kotlin.learn.feature.services.R;

public class MyForegroundService extends Service {
    @Override
    public int onStartCommand(Intent intent, int flags, int startId) {
        new Thread(
                new Runnable() {
                    @Override
                    public void run() {
                        while (true) {
                            Log.e("Service", "Service Foreground is running...");
                            try {
                                Thread.sleep(2000);
                            } catch (InterruptedException e) {
                                e.printStackTrace();
                            }
                        }
                    }
                }
        ).start();

        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.O) {
            final String channelId = "Foreground Service ID";
            NotificationChannel channel;

            channel = new NotificationChannel(
                    channelId,
                    channelId,
                    NotificationManager.IMPORTANCE_LOW
            );
            getSystemService(NotificationManager.class).createNotificationChannel(channel);
            Notification.Builder notification = new Notification.Builder(this, channelId)
                    .setContentText("Service Foreground is running")
                    .setContentTitle("Service Foreground enabled")
                    .setSmallIcon(R.drawable.ic_discover);

            startForeground(1001, notification.build());
        }

        return super.onStartCommand(intent, flags, startId);
    }

    @Nullable
    @Override
    public IBinder onBind(Intent intent) {
        return null;
    }
}

File: feature/services/src/main/java/com/kotlin/learn/feature/services/sample/MyForegroundService.kt
package com.kotlin.learn.feature.services.sample

import android.app.Notification
import android.app.NotificationChannel
import android.app.NotificationManager
import android.app.Service
import android.content.Intent
import android.os.Build
import android.os.IBinder
import com.kotlin.learn.core.utilities.Constant.ONE_HUNDRED_ONE
import com.kotlin.learn.core.utilities.Constant.THREAD_SLEEP_TIMER_2_SECOND
import com.kotlin.learn.feature.services.R

class MyForegroundService : Service() {
    override fun onStartCommand(intent: Intent, flags: Int, startId: Int): Int {
        Thread {
            while (true) {
                try {
                    Thread.sleep(THREAD_SLEEP_TIMER_2_SECOND)
                } catch (e: InterruptedException) {
                    e.printStackTrace()
                }
            }
        }.start()
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            val channelId = "Foreground Service ID"
            val channel = NotificationChannel(
                channelId,
                channelId,
                NotificationManager.IMPORTANCE_LOW
            )
            getSystemService(NotificationManager::class.java).createNotificationChannel(channel)
            val notification = Notification.Builder(this, channelId)
                .setContentText("Service Foreground is running")
                .setContentTitle("Service Foreground enabled")
                .setSmallIcon(R.drawable.ic_discover)
            startForeground(ONE_HUNDRED_ONE, notification.build())
        }
        return super.onStartCommand(intent, flags, startId)
    }

    override fun onBind(intent: Intent): IBinder? {
        return null
    }
}